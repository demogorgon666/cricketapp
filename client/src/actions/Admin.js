import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
