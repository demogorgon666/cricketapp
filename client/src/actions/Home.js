import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
