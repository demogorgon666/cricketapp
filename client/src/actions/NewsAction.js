import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
