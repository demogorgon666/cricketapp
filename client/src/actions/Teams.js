import {
	GET_TEAMS,
	GET_RANKS,
	GET_MATCHBYTEAMID,
	GET_TEAM_BATSMEN,
	GET_TEAM_BOWLERS,
	GET_TEAM_FIXTURES,
	GET_TEAM_STATS
} from "./Types";

import { startLoading, stopLoading } from "./LoadingAction";

import axios from "axios";

export const getTeams = match_type => dispatch => {
	return axios
		.post("http://localhost:5000/cricketalpha/teams", match_type)
		.then(res => {
			dispatch({
				type: GET_TEAMS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getRanks = (gender, ranking) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/rankings/" + gender,
			ranking
		)
		.then(res => {
			dispatch({
				type: GET_RANKS,
				payload: res.data.data
			});
			console.log("from actions", ranking);
		})
		.catch(err => {
			console.log(err);
		});
};

export const getMatch = (team_id, gender, match_type) => dispatch => {
	dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/match/" +
				team_id +
				"/" +
				gender,
			match_type
		)
		.then(res => {
			dispatch(stopLoading());
			dispatch({
				type: GET_MATCHBYTEAMID,
				payload: res.data.data
			});
		})
		.catch(err => {
			dispatch(startLoading());
			console.log(err);
		});
};

export const getTeamBatsmen = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbatsmen/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BATSMEN,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamBowlers = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbowlers/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BOWLERS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

//NIKHIL
export const getFixtures = team_name => dispatch => {
	console.log("get fix from actions", team_name);
	return axios
		.post("http://localhost:5000/cricketalpha/teams/fixtures", team_name)
		.then(res => {
			dispatch({
				type: GET_TEAM_FIXTURES,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamStats = (team_id, gender, type) => dispatch => {
	console.log("getTeamStats action- ", team_id, gender, type);

	// dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/stats/" +
				team_id +
				"/" +
				gender,
			type
		)

		.then(res => {
			// dispatch(stopLoading());
			console.log("getTeamStats Action");
			dispatch({
				type: GET_TEAM_STATS,
				payload: res.data.data
			});
		})
		.catch(err => {
			// dispatch(startLoading());
			console.log(err);
		});
};
