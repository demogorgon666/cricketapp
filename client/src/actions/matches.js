import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

