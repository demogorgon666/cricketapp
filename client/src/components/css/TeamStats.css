.container-team-details {
  margin-left: 100px;
}

.types-section {
  display: flex;
}
.match-type-section {
  width: 50%;
}

.category-section {
  width: 50%;
}

.p-match-type {
  color: #2c3e50;
  font-size: 20px;
  margin-left: 15px;
}

.p-category {
  color: #2c3e50;
  font-size: 20px;
  margin-left: 15px;
}

.p-category-name {
  color: #2c3e50;
  font-size: 24px;
  margin: 50px 0px 25px 12px;
}

.cards-team-stats {
  display: flex;
  margin: 5px 250px 15px 15px;
}

.cards-team-category {
  display: flex;
  margin: 5px 100px 15px 15px;
}

.grid-table {
  /* border: 1px solid #707070; */
  border-radius: 8px;
  margin: 0px 100px 20px 20px;
  background-color: white;
}
.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1.3fr;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin: 0px 10px 0px 10px;
  padding: 1px;
}
.grid-container:not(:last-child) {
  border-bottom: 1px solid #707070;
}

.grid-item-title {
  font-size: 18px;
  font-weight: 500;
  text-align: center;
  color: #2c3e50;
}

.grid-item-detail {
  font-size: 15px;
  text-align: center;
  color: #2c3e50;
}

/* .grid-item-score {
  padding: 10px 10px 10px 0px;
  font-size: 30px;
  text-align: center;
  border-bottom: 1px solid #707070;
  margin-left: 20px;
  font-size: 15px;
  color: #2c3e50;
}
.grid-item-overs {
  padding: 10px 10px 10px 12px;
  font-size: 30px;
  text-align: center;
  border-bottom: 1px solid #707070;
  font-size: 15px;
  color: #2c3e50;
}
.grid-item-rr {
  padding: 10px 10px 10px 20px;
  font-size: 30px;
  text-align: center;
  border-bottom: 1px solid #707070;
  font-size: 15px;
  color: #2c3e50;
}
.grid-item-inns {
  padding: 10px 10px 10px 10px;
  font-size: 30px;
  text-align: center;
  border-bottom: 1px solid #707070;
  font-size: 15px;
  color: #2c3e50;
}
.grid-item-date {
  padding: 10px 10px 10px 35px;
  font-size: 30px;
  text-align: center;
  border-bottom: 1px solid #707070;
  margin-right: 20px;
  font-size: 15px;
  color: #2c3e50;
} */
/* --------------------------------------------------------------------------------- */
@media only screen and (max-width: 1024px) {
}

@media only screen and (max-width: 768px) {
  .cards-team-stats {
    margin: 5px 25px 15px 15px;
  }
  .cards-team-category {
    margin: 5px 7px 15px 15px;
  }
  .container-team-details {
    margin-left: 5px;
  }
  .grid-table {
    margin: 0px 10px 20px 20px;
  }
}

@media only screen and (max-width: 576px) {
  .types-section {
    display: block;
  }
  .cards-team-stats {
    margin: 5px 0px 15px 15px;
  }
  .cards-team-category {
    margin: 5px 15px 15px 15px;
  }
  .container-team-details {
    margin-left: 5px;
  }
  .grid-table {
    margin: auto;
  }
  .grid-item-title {
    font-size: 13px;
    font-weight: 600;
  }
  .grid-item-detail {
    font-size: 13px;
  }
  .grid-container {
    grid-template-columns: 1fr 1fr 1fr 1fr 1.5fr;
  }
}
