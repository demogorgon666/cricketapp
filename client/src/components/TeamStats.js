import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/TeamStats.css";
import { getTeamStats } from "../actions/Teams";
// import Loader from "react-loader-spinner";
// import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class TeamStats extends Component {
  state = {
    testClick: false,
    odiClick: false,
    t20Click: true,
    highestClick: true,
    lowestClick: false,
    match_type: "T20",
    stats_type: "highest_score"
  };

  componentDidMount() {
    console.log("team stats team id is ", this.props.teams.team_id);
    console.log("team stats gender is", this.props.gender);
    console.log("team stats match type is", this.state.match_type);
    console.log("team stats stats type is", this.state.stats_type);
    let type = {
      match_type: this.state.match_type,
      stats_type: this.state.stats_type
    };
    this.props.getTeamStats(this.props.teams.team_id, this.props.gender, type);
  }

  // onClickTest = e => {
  //   e.preventDefault();
  //   this.setState({
  //     testClick: true,
  //     odiClick: false,
  //     t20Click: false,
  //     type: "Test"
  //   });
  // if ((this.state.t20Click = true) && (this.state.lowestClick = false)) {
  // let type = {
  //   match_type: "Test",
  //   gender: "male"
  // };
  // this.onClickHighestTotals;
  // }
  // if ((this.state.odiClick = true)) {
  // let type = {
  //   match_type: "Test"
  // };
  // this.onClickLowestTotals;
  // }
  // this.props.getHighestTotals(this.props.teams.team_id, type);
  // this.props.getLowestTotals(this.props.teams.team_id, this.state.type);
  // this.onClickHighestTotals(e);
  // };

  // onClickOdi = e => {
  //   e.preventDefault();
  //   this.setState({
  //     testClick: false,
  //     odiClick: true,
  //     t20Click: false,
  //     type: "ODI"
  //   });
  // this.props.getHighestTotals(this.props.teams.team_id, type);
  // this.onClickHighestTotals(e);
  // this.props.getHighestTotals(this.props.teams.team_id, this.state.type);
  // this.props.getLowestTotals(this.props.teams.team_id, this.state.type);
  // };

  // onClickT20 = e => {
  //   e.preventDefault();
  //   this.setState({
  //     testClick: false,
  //     odiClick: false,
  //     t20Click: true,
  //     type: "T20"
  //   });
  // this.props.getHighestTotals(this.props.teams.team_id, this.state.type);
  // };

  // onClickHighestTotals = e => {
  //   e.preventDefault();
  //   this.setState({
  //     highestClick: true,
  //     lowestClick: false
  //   });
  //   this.props.getHighestTotals(this.props.teams.team_id, this.state.type);
  // };

  // onClickLowestTotals = e => {
  //   e.preventDefault();
  //   this.setState({
  //     highestClick: false,
  //     lowestClick: true
  //   });
  //   this.props.getLowestTotals(this.props.teams.team_id, this.state.type);
  // };

  onClickTeamStats = (match_type, stats_type) => {
    this.setState({
      match_type: match_type,
      stats_type: stats_type
    });
    let type = {
      match_type: match_type,
      stats_type: stats_type
    };
    this.props.getTeamStats(this.props.teams.team_id, this.props.gender, type);
  };

  render() {
    // console.log("gender for team stats render is", this.props);
    return (
      <div>
        <div className="container-team-details">
          {/* ---------------------------------------------------------------------------- */}
          <div className="types-section">
            <div className="match-type-section">
              <p className="p-match-type">Select Match Type</p>
              <div className="centered-team">
                <div className="cards-team-stats">
                  <div
                    style={{ borderRadius: "8px 0px 0px 8px" }}
                    className={
                      this.state.match_type == "T20" ? "cardtest" : "cardt20"
                    }
                    onClick={() =>
                      this.onClickTeamStats("T20", this.state.stats_type)
                    }
                  >
                    <p className="p-card">T20</p>
                  </div>
                  <div
                    className={
                      this.state.match_type == "ODI" ? "cardtest" : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats("ODI", this.state.stats_type)
                    }
                  >
                    <p className="p-card">ODI</p>
                  </div>
                  <div
                    style={{ borderRadius: "0px 8px 8px 0px" }}
                    className={
                      this.state.match_type == "Test" ? "cardtest" : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats("Test", this.state.stats_type)
                    }
                  >
                    <p className="p-card">Test</p>
                  </div>
                </div>
              </div>
            </div>
            <div className="category-section">
              <p className="p-category">Select Category</p>
              <div className="centered-team">
                <div className="cards-team-category">
                  <div
                    style={{ borderRadius: "8px 0px 0px 8px" }}
                    className={
                      this.state.stats_type == "highest_score"
                        ? "cardtest"
                        : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats(
                        this.state.match_type,
                        "highest_score"
                      )
                    }
                  >
                    <p className="p-card">Highest Totals</p>
                  </div>
                  <div
                    className={
                      this.state.stats_type == "lowest_score"
                        ? "cardtest"
                        : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats(
                        this.state.match_type,
                        "lowest_score"
                      )
                    }
                  >
                    <p className="p-card">Lowest Totals</p>
                  </div>
                  <div
                    className="cardodi"
                    className={
                      this.state.stats_type == "largest_victory"
                        ? "cardtest"
                        : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats(
                        this.state.match_type,
                        "largest_victory"
                      )
                    }
                  >
                    <p className="p-card">Largest Victories</p>
                  </div>
                  <div
                    style={{ borderRadius: "0px 8px 8px 0px" }}
                    className={
                      this.state.stats_type == "smallest_victory"
                        ? "cardtest"
                        : "cardodi"
                    }
                    onClick={() =>
                      this.onClickTeamStats(
                        this.state.match_type,
                        "smallest_victory"
                      )
                    }
                  >
                    <p className="p-card">Smallest Victories</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          {/* ------------------------------------------------------------------------- */}
          {/* {this.props.isLoading ? (
            <div style={{ margin: "auto" }}>
              <Loader
                type="TailSpin"
                color="#2980b9"
                height="100"
                width="100"
              />
            </div>
          ) : ( */}
          <div className="stats-display">
            <p className="p-category-name">
              {this.state.stats_type == "largest_victory"
                ? "Largest Victories"
                : this.state.stats_type == "smallest_victory"
                ? "Smallest Victories"
                : this.state.stats_type == "highest_score"
                ? "Highest Totals"
                : "Lowest Totals"}
            </p>
            {console.log(this.props.teamstats)}
            <div className="grid-table">
              <div className="grid-container">
                <p className="grid-item-title">
                  {this.props.teamstats.length == 0
                    ? null
                    : this.props.teamstats[0].outcome
                    ? "Margin"
                    : "Score"}
                </p>
                <p className="grid-item-title">Overs</p>
                <p className="grid-item-title">Opposition</p>
                <p className="grid-item-title">Inns</p>
                <p className="grid-item-title">MatchDate</p>
              </div>
              {this.props.teamstats.map(teamstats => (
                <div className="grid-container">
                  <p className="grid-item-detail">
                    {teamstats.outcome
                      ? teamstats.outcome
                      : teamstats.total_run}
                  </p>
                  <p className="grid-item-detail">{teamstats.overs}</p>
                  <p className="grid-item-detail">
                    {teamstats.team_one} vs {teamstats.team_two}
                  </p>
                  <p className="grid-item-detail">{teamstats.inning}</p>
                  <p className="grid-item-detail">{teamstats.match_date}</p>
                </div>
              ))}
            </div>
          </div>
          {/* )} */}
        </div>
      </div>
    );
  }
}

const mapStateTostate = state => ({
  teamstats: state.TeamsReducer.teamstats
});

export default connect(
  mapStateTostate,
  // { getHighestTotals, getLowestTotals }
  { getTeamStats }
)(TeamStats);
