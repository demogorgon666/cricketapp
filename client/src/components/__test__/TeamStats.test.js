import React from "react";
import { shallow } from "enzyme";
import { TeamStats } from "../TeamStats";
import { getTeamStats } from "../../actions/Teams";

var teams = [],
  teamstats = [];

const wrapper = shallow(
  <TeamStats getTeamStats={getTeamStats} teamstats={teamstats} teams={teams} />
);

describe("Test TeamStats Component", () => {
  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have text for match type", () => {
    expect(wrapper.find("#p-stats-match-type").text()).toBe(
      "Select Match Type"
    );
  });
  it("tab should have text for T20", () => {
    expect(wrapper.find("#p-stats-match-type-t20").text()).toBe("T20");
  });
  it("tab should have text for ODI", () => {
    expect(wrapper.find("#p-stats-match-type-odi").text()).toBe("ODI");
  });
  it("tab should have text for Test", () => {
    expect(wrapper.find("#p-stats-match-type-test").text()).toBe("Test");
  });
  it("should have text for match type", () => {
    expect(wrapper.find("#p-stats-category").text()).toBe("Select Category");
  });
  it("tab should have text for Highest Totals", () => {
    expect(wrapper.find("#p-stats-category-ht").text()).toBe("Highest Totals");
  });
  it("tab should have text for Lowest Totals", () => {
    expect(wrapper.find("#p-stats-category-lt").text()).toBe("Lowest Totals");
  });
  it("tab should have text for Largest Margin", () => {
    expect(wrapper.find("#p-stats-category-lv").text()).toBe(
      "Largest Victories"
    );
  });
  it("tab should have text for Smallest Margin", () => {
    expect(wrapper.find("#p-stats-category-sv").text()).toBe(
      "Smallest Victories"
    );
  });

  // it("should change the state of testClicked to true when test tab is clicked", () => {
  //   const test = wrapper.find("#click-test");
  //   test.simulate("click");
  //   expect(wrapper.state().testClick).toBe(true);
  //   expect(wrapper.state().odiClick).toBe(false);
  //   expect(wrapper.state().t20Click).toBe(false);
  // });

  // it("should change the state of odiclicked to true when odi tab is clicked", () => {
  //   const odi = wrapper.find("#click-odi");
  //   odi.simulate("click");
  //   expect(wrapper.state().testClick).toBe(false);
  //   expect(wrapper.state().odiClick).toBe(true);
  //   expect(wrapper.state().t20Click).toBe(false);
  // });

  // it("should change the state of t20Click to true when IT20 tab is clicked", () => {
  //   const t20 = wrapper.find("#click-t20");
  //   t20.simulate("click");
  //   expect(wrapper.state().odiClick).toBe(false);
  //   expect(wrapper.state().testClick).toBe(false);
  //   expect(wrapper.state().t20Click).toBe(true);
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-match-type").text()).toBe("ODI");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-team-one").text()).toBe("South Africa");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-team-two").text()).toBe("Australia");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-winner").text()).toBe("South Africa WON");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-batsman-name").text()).toBe("Ms Dhoni");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-batsman-score").text()).toBe("10004");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-bowler-name").text()).toBe("Mohammad Sami");
  // });
  // it("Match type should be shown", () => {
  //   expect(wrapper.find("#id-bowler-score").text()).toBe("15");
  // });
  it("should have no br tags", () => {
    expect(wrapper.find("br").length).toBe(0);
  });
});
