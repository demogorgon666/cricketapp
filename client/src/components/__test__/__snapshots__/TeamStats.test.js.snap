// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TeamStats Component render the component 1`] = `
<div>
  <div
    className="container-team-details"
  >
    <div
      className="types-section"
    >
      <div
        className="match-type-section"
      >
        <p
          className="p-match-type"
          id="p-stats-match-type"
        >
          Select Match Type
        </p>
        <div
          className="centered-team"
        >
          <div
            className="cards-team-stats"
          >
            <div
              className="cardtest"
              onClick={[Function]}
              style={
                Object {
                  "borderRadius": "8px 0px 0px 8px",
                }
              }
            >
              <p
                className="p-card"
                id="p-stats-match-type-t20"
              >
                T20
              </p>
            </div>
            <div
              className="cardodi"
              onClick={[Function]}
            >
              <p
                className="p-card"
                id="p-stats-match-type-odi"
              >
                ODI
              </p>
            </div>
            <div
              className="cardodi"
              onClick={[Function]}
              style={
                Object {
                  "borderRadius": "0px 8px 8px 0px",
                }
              }
            >
              <p
                className="p-card"
                id="p-stats-match-type-test"
              >
                Test
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="category-section"
      >
        <p
          className="p-category"
          id="p-stats-category"
        >
          Select Category
        </p>
        <div
          className="centered-team"
        >
          <div
            className="cards-team-category"
          >
            <div
              className="cardtest"
              onClick={[Function]}
              style={
                Object {
                  "borderRadius": "8px 0px 0px 8px",
                }
              }
            >
              <p
                className="p-card"
                id="p-stats-category-ht"
              >
                Highest Totals
              </p>
            </div>
            <div
              className="cardodi"
              onClick={[Function]}
            >
              <p
                className="p-card"
                id="p-stats-category-lt"
              >
                Lowest Totals
              </p>
            </div>
            <div
              className="cardodi"
              onClick={[Function]}
            >
              <p
                className="p-card"
                id="p-stats-category-lv"
              >
                Largest Victories
              </p>
            </div>
            <div
              className="cardodi"
              onClick={[Function]}
              style={
                Object {
                  "borderRadius": "0px 8px 8px 0px",
                }
              }
            >
              <p
                className="p-card"
                id="p-stats-category-sv"
              >
                Smallest Victories
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="stats-display"
    >
      <p
        className="p-category-name"
      >
        Highest Totals
      </p>
      <div
        className="grid-table"
      >
        <div
          className="grid-container"
        >
          <p
            className="grid-item-title"
          />
          <p
            className="grid-item-title"
          >
            Overs
          </p>
          <p
            className="grid-item-title"
          >
            Opposition
          </p>
          <p
            className="grid-item-title"
          >
            Inns
          </p>
          <p
            className="grid-item-title"
          >
            MatchDate
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
